Documentation/migration-scripts/python-scripts/push_kafka_messages.py

```
@@ -0,0 +1,298 @@
+        cursor = conn.cursor(cursor_factory=psycopg2.extras.RealDictCursor)
+
+        # === Fetch and Process project-linked dashboard ===
+        log("=== Fetch and Process project-linked dashboard ===")
+        cursor.execute(f"""
+            SELECT linked_to, entity_id
+            FROM {TABLE_NAME}
+            WHERE entity_type = 'solution'
+              AND report_type = 'improvementProject'
+              AND linked_to IS NOT NULL AND linked_to <> 'null'
+              AND entity_id IS NOT NULL AND entity_id <> 'null';
+        """)
```

Ruff (0.15.1)

[error] Possible SQL injection vector through string-based query construction

(S608)